src/callbacks_step/timeseries.jl

```diff
@@ -0,0 +1,202 @@
+function Base.show(io::IO, ::MIME"text/plain", cb::DiscreteCallback{<:Any, <:TimeseriesCallback})
+  @nospecialize cb # reduce precompilation time
+
+  if get(io, :compact, false)
+    show(io, cb)
+  else
+    timeseries_callback = cb.affect!
+
+    setup = [
+             "#points" => size(timeseries_callback.point_coordinates, 2),
+             "interval" => timeseries_callback.interval,
+             "solution_variables" => timeseries_callback.solution_variables,
+             "output_directory" => timeseries_callback.output_directory,
+             "filename" => timeseries_callback.filename,
+            ]
+    summary_box(io, "TimeseriesCallback", setup)
+  end
+end
+
+
+# Main constructor
+function TimeseriesCallback(mesh, equations, solver, cache, point_coordinates;
+                            interval::Integer=1,
+                            solution_variables=cons2cons,
+                            output_directory="out",
+                            filename="timeseries.h5",
+                            RealT=real(solver),
+                            uEltype=eltype(cache.elements))
+
+  # check arguments
+  if !(interval isa Integer && interval >= 0)
+    throw(ArgumentError("`interval` must be a non-negative integer (provided `interval = $interval`)"))
+  end
+
+  # Invoke callback every `interval` time steps or after final step (for storing the data on disk)
+  if interval > 0
+    condition = (u, t, integrator) -> (integrator.iter % interval == 0 || isfinished(integrator))
+  else # disable the callback for interval == 0
+    condition = (u, t, integrator) -> false
+  end
+
+  # Create data structures that are to be filled by the callback
+  variable_names = varnames(solution_variables, equations)
+  n_points = size(point_coordinates, 2)
+  point_data = Vector{uEltype}[Vector{uEltype}() for _ in 1:n_points]
+  time = Vector{RealT}()
+  step = Vector{Int}()
+  timeseries_cache = create_cache_timeseries(point_coordinates, mesh, solver, cache)
+
+  timeseries_callback = TimeseriesCallback(interval,
+                                           solution_variables,
+                                           variable_names,
+                                           output_directory,
+                                           filename,
+                                           point_coordinates,
+                                           point_data,
+                                           time,
+                                           step,
+                                           timeseries_cache)
+
+  return DiscreteCallback(condition, timeseries_callback, save_positions=(false,false))
+end
+
+
+# Convenience constructor that unpacks the semidiscretization into mesh, equations, solver, cache
+function TimeseriesCallback(semi, point_coordinates; kwargs...)
+  mesh, equations, solver, cache = mesh_equations_solver_cache(semi)
+
+  return TimeseriesCallback(mesh, equations, solver, cache, point_coordinates; kwargs...)
+end
```

Comment:
It's slightly confusing for me that the position of `point_coordinates` is not consistent

Reply:
Hmm, it's fine as it is, I guess. Never mind.